From: Alexis Murzeau <amubtdx@gmail.com>
Date: Mon, 2 May 2022 21:40:05 +0200
Subject: Remove dependency on versioningit

The version string is already generated in the distributed source
tarball.

Forwarded: not-needed
Author: Alexis Murzeau <amubtdx@gmail.com>
---
 dev-requirements.txt  | 1 -
 docs-requirements.txt | 1 -
 pyproject.toml        | 1 -
 setup.cfg             | 1 +
 setup.py              | 2 --
 5 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/dev-requirements.txt b/dev-requirements.txt
index 6b1c7d5..1f5e3a9 100644
--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -7,7 +7,6 @@ freezegun>=1.0.0
 flake8
 flake8-import-order
 shtab
-versioningit >=1.1.1, <2
 
 mypy
 lxml-stubs
diff --git a/docs-requirements.txt b/docs-requirements.txt
index b5c2bdc..eb31ad9 100644
--- a/docs-requirements.txt
+++ b/docs-requirements.txt
@@ -1,4 +1,3 @@
 sphinx>=3.0
 furo==2021.09.08
 recommonmark>=0.5.0
-versioningit >=1.1.1, <2
diff --git a/pyproject.toml b/pyproject.toml
index 9901699..b22981f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,7 +3,6 @@ build-backend = "setuptools.build_meta"
 requires = [
   "setuptools >=45",
   "wheel",
-  "versioningit >=1.1.1, <2",
 ]
 
 
diff --git a/setup.cfg b/setup.cfg
index da89da4..ac5b55b 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,5 +1,6 @@
 [metadata]
 name = streamlink
+version = attr:streamlink.__version__
 author = Streamlink
 author_email = streamlink@protonmail.com
 description = Streamlink is a command-line utility that extracts streams from various services and pipes them into a video player of choice.
diff --git a/setup.py b/setup.py
index 515c145..f638edf 100644
--- a/setup.py
+++ b/setup.py
@@ -74,10 +74,8 @@ data_files = [
 
 if __name__ == "__main__":
     from setuptools import setup
-    from versioningit import get_cmdclasses
 
     setup(
-        cmdclass=get_cmdclasses(),
         entry_points=entry_points,
         data_files=data_files,
     )
